docs/src/main/asciidoc/building-my-first-extension-guide.adoc

```diff
@@ -0,0 +1,737 @@
+
+Go back to the generated `org.acme.quarkus.greeting.deployment.GreetingProcessor` class.
+
+[source, java]
+----
+package org.acme.quarkus.greeting.deployment;
+
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.builditem.FeatureBuildItem;
+
+class GreetingProcessor {
+
+    private static final String FEATURE = "greeting";
+
+    @BuildStep // <1>
+    FeatureBuildItem feature() {
+        return new FeatureBuildItem(FEATURE); // <2>
+    }
+
+}
+----
+
+<1> `feature()` method is annotated with `@BuildStep` which means it is identified as a deployment task Quarkus will have to execute during the deployment.
+
+`BuildStep` methods are run concurrently at augmentation time to augment the application.
+They use a producer/consumer model, where a step is guaranteed not to be run until all the items that it is consuming have been produced.
+
+<2> `io.quarkus.deployment.builditem.FeatureBuildItem` is an implementation of `BuildItem` which represents the description of an extension.
+This `BuildItem` will be used by Quarkus to display information to the users when the application is starting.
+
+There are many `BuildItem` implementations, each one represents an aspect of the deployment process.
+Here are some examples:
+
+* `ServletBuildItem`: describes a Servlet (name, path, etc.) we want to generate during the deployment.
+* `BeanContainerBuildItem`: describes a container used to store and retrieve object instances during the deployment.
+
+If you don't find a `BuildItem` for what you want to achieve, you can create your own implementation. Keep in mind that a `BuildItem` should be as fine-grained as possible, representing a specific part of the deployment.
+To create your `BuildItem` you can extend:
+
+* `io.quarkus.builder.item.SimpleBuildItem` if you need only a single instance of the item during the deployment (e.g. `BeanContainerBuildItem`, you only want one container).
+* `io.quarkus.builder.item.MultiBuildItem` if you want to have multiple instances (e.g. `ServletBuildItem`, you can produce many Servlets during the deployment).
+
+It's now time to declare our HTTP endpoint. To do so, we need to produce a `ServletBuildItem`.
+At this point, we are sure you understood that if the `quarkus-undertow` dependency proposes Servlet support for our `runtime` module, we will need the `quarkus-undertow-deployment` dependency into our `deployment` module to have access to the `io.quarkus.undertow.deployment.ServletBuildItem`.
+
+Update the `./greeting/deployment/pom.xml` as follows:
+
+[source, xml]
+----
+    <dependencies>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-core-deployment</artifactId>
+        </dependency>
```

Comment:
BTW, direct dependency on quarkus-core-deployment isn't necessary once quarkus-undertow-deployment has been added (it gets pulled in as a transitive dep instead).

Reply:
Good point. The core deployment comes from the MOJO it could be misleading for newcomers